builder/lxc/step_lxc_create.go

```diff
@@ -14,54 +16,90 @@ import (
 
 type stepLxcCreate struct{}
 
+func (s *stepLxcCreate) createFromTemplate(containerName string, config LxcTemplateConfig) (string, error) {
+	lxcDir := "/var/lib/lxc"
+	rootfs := filepath.Join(lxcDir, containerName, "rootfs")
+
+	commands := make([][]string, 2)
+	commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", containerName, "-t", config.Name, "--"}...)
+	commands[0] = append(commands[0], config.Parameters...)
+	// prevent tmp from being cleaned on boot, we put provisioning scripts there
+	// TODO: wait for init to finish before moving on to provisioning instead of this
+	commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
+
+	err := s.SudoCommands(commands...)
+	return rootfs, err
+}
+
+func (s *stepLxcCreate) createFromRootFs(containerName string, config RootFsConfig) (string, error) {
+	lxcDir := "/var/lib/lxc"
+	containerPath := filepath.Join(lxcDir, containerName)
+	rootfs := filepath.Join(containerPath, "rootfs")
+	containerConfig, err := NewLxcConfig(config.ConfigFile)
+	if err != nil {
+		err = fmt.Errorf("Could not read lxc config (%s): %s", config.ConfigFile, err)
+		return "", err
+	}
+	containerConfig.SetRootFs(rootfs)
+	tmpDir, err := ioutil.TempDir("", "lxcconfig")
+	if err != nil {
+		err = fmt.Errorf("Could not create temp directory for lxc config (%s): %s", tmpDir, err)
+		return rootfs, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = containerConfig.Write(filepath.Join(tmpDir, "lxc.config"))
```

Comment:
`/var/lib/lxc` is only writable by root. That's the only reason why I'm writing to a temp dir first before using `SudoCommand` later to do the copy. Ultimately, I'd like to use `--lxcPath` to do all of this in an arbitrary (and thus user writable) directory instead of the system defined lxc directory at `/var/lib/lxc`.